src/styles/blocks/day.scss

```diff
@@ -0,0 +1,30 @@
+.day {
+  position: relative;
+  box-sizing: border-box;
+  width: $width-cell;
+  height: 100px;
+  border: 1px solid black;
+  background-color: #eee;
+  transition: background-color 0.5s, transform 0.5s;
+
+  @for $dayCount from 1 through $day-amount {
+    &--#{$dayCount}::before {
+      content: "#{$dayCount}";
+      position: absolute;
+      left: 0;
+      right: 0;
+      top: 0;
+      bottom: 0;
+      font-family: Arial, Helvetica, sans-serif;
+      text-align: center;
+      line-height: 100px;
+      font-size: 30px;
```

Comment:
no need to repeat all styles in a loop
you can change only `content`
other styles are static, so move them to `day` class
<img width="796" alt="image" src="https://user-images.githubusercontent.com/102411130/236754236-7d0c08f0-4cf7-49d4-a4d3-86da3e468c1b.png">